pkgs/development/libraries/tomlplusplus/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib
+, stdenv
+, meson
+, ninja
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tomlplusplus";
+  version = "3.2.0";
+
+  src = fetchFromGitHub {
+    owner = "marzer";
+    repo = pname;
+    rev = "4b166b69f28e70a416a1a04a98f365d2aeb90de8";
+    hash = "sha256-nohO4eySs73BSgjvq+uzybiE5lw2rFY5YqGbl/oqGek=";
+  };
+
+  nativeBuildInputs = [ meson ninja ];
+
+  meta = with lib;{
+    homepage = "https://github.com/marzer/tomlplusplus";
+    description = "Header-only TOML config file parser and serializer for C++17";
+    license = licenses.mit;
+    maintainers = with maintainers; [ Scrumplex ];
```

Comment:
It's unusual to add someone else as a maintainer, unless you're also @Scrumplex on a different account?

Reply:
Oh my bad, this is from Scrumplex's PR and not for merging.

Reply:
I mentioned this in the PR description: these are cherry-picked changes for #193038, which needs to be merged for this PR to work properly. I've picked them into this branch so that people can test properly, although I do not think this should be merged until I have dropped them from here and rebased onto a version of master that has them in